v0/destinations/google_ads_enhanced_conversion/transform.js

```diff
@@ -0,0 +1,119 @@
+const {
+  getSuccessRespEvents,
+  getErrorRespEvents,
+  CustomError,
+  constructPayload,
+  defaultRequestConfig,
+  getValueFromMessage
+} = require("../../util");
+
+const { trackMapping, BASE_ENDPOINT } = require("./config");
+
+const responseBuilder = async (metadata, message, { Config }, payload) => {
+  const response = defaultRequestConfig();
+  const { event } = message;
+  let flag = 0;
+  const { listOfConversions } = Config;
+  for (let i = 0; i < listOfConversions.length; i += 1) {
+    if (listOfConversions[i].conversions === event) {
+      flag = 1;
+      break;
+    }
+  }
+  if (event === undefined || event === "" || flag === 0) {
+    throw new CustomError(
+      `[Google_ads_enhanced_marketing]:: Conversion named ${event} is not exist in rudderstack dashboard`,
+      400
+    );
+  }
+
+  response.endpoint = `${BASE_ENDPOINT}/${Config.customerId}:uploadConversionAdjustments`;
+  response.body.JSON = payload;
+  const accessToken = metadata.secret.access_token;
+  response.headers = {
+    Authorization: `Bearer ${accessToken}`,
+    "Content-Type": "application/json",
+    "developer-token": getValueFromMessage(metadata, "secret.developer_token")
+  };
+  response.params = { event, customerId: Config.customerId };
+  if (Config.subAccount)
+    if (Config.loginCustomerId)
+      response.headers["login-customer-id"] = Config.loginCustomerId;
+    else
+      throw new CustomError(
+        `[Google_ads_enhanced_marketing]:: loginCustomerId is required as subAccount is true.`,
+        400
+      );
+  return response;
+};
+
+const processTrackEvent = async (metadata, message, destination) => {
+  const payload = constructPayload(message, trackMapping);
+  if (!payload.conversionAdjustments[0].userIdentifiers) {
```

Comment:
should we put a check if `conversionAdjustments` exists on `payload` and if it is of type `Array` etc ?

Reply:
We don't need to check this because the payload will always get created in that way.

Reply:
are we sure on that? asking because I did not see any `required: true` for the fields in the mapping json. what if they are not there?

Reply:
ok. we would probably need to refactor this part

Reply:
orderId is required as true under conversionAdjustments[0].